### PR TITLE
Update to swap Gitness with Harness Open Source.

### PR DESCRIPTION
**Pull request by dewan-ahmed**
This PR updates some `Gitness` references with `Harness Open Source`.

https://github.com/harness/developer-hub/pull/8173 is open to add /docs/open-source CODEOWNERS. Till that PR is merged, the code contributions to /docs/open-source will go to overall docs owners.

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/open-source/installation/data.md [hunk 1]
```diff
@@ -12,43 +12,43 @@ When running a shared Harness Open Source instance for a development team, you s
...
 
 Create a separate volume just for your Harness Open Source instance data. Ensure the volume is an appropriate size for your team and number of repositories.
 
-Create an [EBS volume](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-using-volumes.html) for your Harness Open Source instance data and mount it at `/mnt/gitness-data`, then pass `-v /mnt/gitness-data:/data` in your `docker run` command.
+Create an [EBS volume](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-using-volumes.html) for your Harness Open Source instance data and mount it at `/mnt/harness-data`, then pass `-v /mnt/harness-data:/data` in your `docker run` command.
 
 ```sh {4} showLineNumbers
 docker run -d \
-  -p 3000:3000 \
+  -p 3000:3000 -p 3022:3022 \
```
Comment (SushrutHarness) [inline docs/open-source/installation/data.md:43]: What is this port change?

File: docs/open-source/overview.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Here’s a short overview of the capabilities:
 - Continuous Integration/Continuous Delivery - Build, test, and deploy code faster with hundreds of reusable pipeline templates.
 - Artifact Registry - Centralize and streamline software delivery pipelines by providing a single source of truth for software artifacts.
 
-## How is Gitness related to Drone?
+## How is Harness Open Source related to Drone?
 
 Gitness was the next step in the evolution of [Drone](https://www.drone.io), from continuous integration to source code hosting, bringing code management and pipelines closer together.
```
Comment (SushrutHarness) [inline docs/open-source/overview.md:18]:
```suggestion
Gitness was the next step in the evolution of [Drone](https://www.drone.io), from continuous integration to source code hosting, bringing code management and pipelines closer together. 

Gitness then became Harness Open Source, further enhancing the open source experience bringing artifact management and gitspaces. 
```

**Review by SushrutHarness — DISMISSED** [review]
One missed change and one question but otherwise LGTM

File: docs/open-source/installation/data.md (hunk 1 above)
Comment (dewan-ahmed) [reply docs/open-source/installation/data.md:43]: To start Harness Open Source with SSH support.

**Review by wei-harness — APPROVED** [review]
LGTM from a code standpoint

**Comment by dewan-ahmed** [comment]
Merging based on https://github.com/harness/developer-hub/pull/8175#pullrequestreview-2341192004

**Pull request merged**